### PR TITLE
IDEMPIERE-4485 Info Window Columns has no Value preference

### PR DESCRIPTION
**Pull request by igorpojzl**
Problem is that Info Column has ReadOnly parameter which have 2 uses. So if we set Parameter as ReadOnly, this parameter is in table set as read only. But if this parameter is set as Query Parameter it will interfere with menu popup which checks ReadOnly Parameter.
So i created parameterTree to store new Gridfield copy which have forced read only false in Query Parameter.

https://idempiere.atlassian.net/browse/IDEMPIERE-4485

### CODE REVIEW

**Comment by igorpojzl** [comment]
- [ ] TODO Filter in InfoWindow is Broken. Wait for Fix.

**Comment by hengsin** [comment]
hi @igorpojzl , is that change set complete ? I didn't see any additions that will show popup/context menu for info window query/parameter field.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java [hunk 1]
```diff
@@ -624,6 +626,40 @@ protected boolean loadInfoDefinition() {
...
+					if (infoColumn.getDisplayLogic() != null)					
+						vo.DisplayLogic =  infoColumn.getDisplayLogic();
+					if (infoColumn.isQueryCriteria() && infoColumn.getDefaultValue() != null)
+						vo.DefaultValue = infoColumn.getDefaultValue();
+					desc = infoColumn.getDescriptionTrl();
+					vo.Description = desc != null ? desc : "";
+					help = infoColumn.getHelpTrl();
+					vo.Help = help != null ? help : "";
+					vo.AD_FieldStyle_ID = infoColumn.getAD_FieldStyle_ID();
+					vo.IsAutocomplete = infoColumn.isAutocomplete();
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java:652]: the code before vo.IsReadOnly can be shorten to the following:
vo = vo.clone(infoContext, p_WindowNo, 0, vo.AD_Window_ID, 0, false);

**Review by hengsin — CHANGES_REQUESTED** [review]
tested working but it can be shorten quite a bit.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java (hunk 1 above)
Comment (igorpojzl) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java]: Commited changes

**Review by hengsin — APPROVED** [review]
tested

**Comment by hengsin** [comment]
I wrote above "the code before vo.IsReadOnly can be shorten to the following", didn't notice @igorpojzl has remove the vo.IsReadOnly = false line as well. that's needed.

**Comment by igorpojzl** [comment]
Oh, my bad sorry, Ive missread the clone method. I thought it set IsReadOnly parameter.

**Pull request merged**